pkg/cache/snapshot_test.go

```diff
@@ -227,23 +227,14 @@ func TestSnapshot(t *testing.T) {
 							ResourceGroups: []ResourceGroup{
 								{
 									CoveredResources: sets.New(corev1.ResourceCPU),
-									Flavors: []FlavorQuotas{
-										{
-											Name: "demand",
-											Resources: map[corev1.ResourceName]*ResourceQuota{
-												corev1.ResourceCPU: {Nominal: 100_000},
-											},
-										},
-										{
-											Name: "spot",
-											Resources: map[corev1.ResourceName]*ResourceQuota{
-												corev1.ResourceCPU: {Nominal: 200_000},
-											},
-										},
-									},
-									LabelKeys: sets.New("instance"),
+									Flavors:          []kueue.ResourceFlavorReference{"demand", "spot"},
+									LabelKeys:        sets.New("instance"),
 								},
 							},
+							Quotas: map[resources.FlavorResource]*ResourceQuota{
```

Comment:
It seems that with this change Quotas are independent of ResourceGroups (at least at this level of abstraction). I'm wondering if this is desired since the API seems to be more expressive. Can you explain if the internal structures have the same expressiveness as the API structures, and if not, are we losing some use cases (like multiple overlapping ResourceGroups)?

Reply:
The API requires each flavor and each resource to belong to at most 1 resource group. As long as that doesn't change, this structure is expressive enough https://github.com/kubernetes-sigs/kueue/blob/542b90cfe294e41b0d677b2606fd2485c704b48a/apis/kueue/v1beta1/clusterqueue_types.go#L31

Reply:
I see, this is the important bit I wasn't sure about. It means that ResourceGroups are disjoint for the underlying flavors and resources, so we don't need to aggregate the values in any way.